OpenRA.Mods.Common/Traits/Player/ProvidesPrerequisite.cs

```diff
@@ -64,19 +63,37 @@ public IEnumerable<string> ProvidesPrerequisites
 
 		public void OnOwnerChanged(Actor self, Player oldOwner, Player newOwner)
 		{
-			if (info.ResetOnOwnerChange)
+			if (Info.ResetOnOwnerChange)
 				Update(newOwner, newOwner.Faction.InternalName);
 		}
 
 		void Update(Player owner, string faction)
 		{
+			if (IsTraitDisabled)
```

Comment:
Minor nit: this can simplify down to
```
enabled = !IsTraitDisabled;
if (!enabled) // or maybe if (IsTraitDisabled) would be clearer?
	return;
```